devtools/limitapigen/main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/target/goalert/limit"
+)
+
+type SystemLimitInfo struct {
+	ID          limit.ID
+	Description string
+	Value       int
+}
+
+type SystemLimitScanner struct {
+	data []SystemLimitInfo
+}
+
+func (s *SystemLimitScanner) Visit(node ast.Node) ast.Visitor {
+	v, ok := node.(*ast.ValueSpec)
+	if !ok {
+		return s
+	}
+
+	name, ok := v.Type.(*ast.Ident)
+	if !ok || name.Name != "ID" {
+		return s
+	}
+
+	var comments []string
+	if v.Doc != nil {
+		comments = make([]string, len(v.Doc.List))
+		for index, val := range v.Doc.List {
+			comments[index] = strings.TrimSpace(strings.TrimPrefix(val.Text, "//"))
+		}
+	}
+
+	s.data = append(s.data, SystemLimitInfo{
+		ID:          limit.ID(v.Names[0].Name),
+		Description: strings.Join(comments, "\n"),
+	})
+
+	return s
+}
+
+var tmpl = template.Must(
+	template.
+		New("maplimit.go").
+		Funcs(template.FuncMap{
+			"quote": strconv.Quote,
+		}).Parse(`
+// Code generate by devtools/limitapigen DO NOT EDIT.
```

Comment:
Should appear as the first line, and generate -> generated

```suggestion
		}).Parse(`// Code generated by devtools/limitapigen DO NOT EDIT.
```